lib/Onyx.ts

```diff
@@ -99,6 +99,11 @@ function connect<TKey extends OnyxKey>(connectOptions: ConnectOptions<TKey>): nu
     callbackToStateMapping[connectionID] = mapping as Mapping<OnyxKey>;
     callbackToStateMapping[connectionID].connectionID = connectionID;
 
+    // When keyChanged is called, the key is passed and looks through all the callbackToStateMapping the right key
+    // to avoid having to loop through all the keys all the time (even when just one connection belongs to one key),
+    // we store the key by connectionID, to access the specific list of connectionIDs
```

Comment:
@gedu sorry can you please try to rewrite this a bit more, I am not sure if I am following, especially this part does not seem correct to me `the key is passed and looks through all the callbackToStateMapping the right key`.

What `looks through` callbackToStateMapping? What `the right key`?

Thanks!